vtysh/vtysh.c

```diff
@@ -1335,6 +1355,39 @@ DEFUNSH(VTYSH_REALLYALL, vtysh_end_all, vtysh_end_all_cmd, "end",
 	return vtysh_end();
 }
 
+DEFUNSH(VTYSH_SR, segment_routing, segment_routing_cmd,
+	"segment-routing",
```

Comment:
Not sure that the 'segment-routing' CLI node is appropriate. Segment Routing with MPLS data plane is managed at the OSPF and ISIS level not at the Zebra level. If such commands are mandatory for SRv6, I would suggest to change the CLI Node and use directly 'segment-routing-v6' or 'srv6' to avoid confusion with segment-routing on top of MPLS data plane.

Reply:
I think locators will need to be configured independent of the control plane in use, so it's okay to add a new Zebra level command here. OTOH, I agree `segment-routing` is going to cause confusion in the future. I'm not even certain we can use `srv6` here, as we are likely to need to use locators for SRm6, etc. So we might need to rethink this a little.

Reply:
> Segment Routing with MPLS data plane is managed at the OSPF and ISIS level not at the Zebra level.

In major NOS and the principle of SRv6, This is well difference between SR-MPLS and SRv6.
As riw777 says, SRv6 locator is independent mechanism for each routing-protocols.

> OTOH, I agree segment-routing is going to cause confusion in the future. I'm not even certain we can use srv6 here, as we are likely to need to use locators for SRm6, etc. So we might need to rethink this a little.

At this time, I'm respecting cisco's CLI format. And I use the same cli format as well cisco.
For the considering SRm6 and more, why don't you thinking after SRm6 is coming?

However,
If we will facing confusion about Segment-Routing XXX (XX is everything), I think `segment-routing xxx` is the best way to separate each segment-routing variation.

### Ref:
- [cisco's SRv6 configuration](https://www.cisco.com/c/en/us/td/docs/routers/asr9000/software/asr9k-r6-6/segment-routing/configuration/guide/b-segment-routing-cg-asr9000-66x/b-segment-routing-cg-asr9000-66x_chapter_011.html)
- [cisco's SR-MPLS configuration](https://www.cisco.com/c/en/us/td/docs/iosxr/ncs5000/segment-routing/b-segment-routing-cr-ncs5000/b-segment-routing-cr-ncs5000_chapter_01.html#wp1899834244)

Reply:
I think using segment-routing might be better here, rather than srv6, unless we want to have one set of commands for each variant (?)... Maybe we need to discuss in the community to figure out what sort of consensus we can reach on this, and to make certain no-one is surprised when this goes in. :-)

Reply:
After thinking about it, `segment-routing` is okay... I wouldn't tie anything to a protocol unless it's directly related to the protocol.

Reply:
Yeap. If Cisco CLI using it ... we could consider the same CLI to let people familiar with a well-know CLI.